### PR TITLE
Category page

### PR DESCRIPTION
**Pull request by birdylockstep**
Category list:
screen shot 2560-04-11 at 1 48 35 pm

When click on any category link (secondary category), it will go to search result page:
screen shot 2560-04-11 at 1 48 47 pm

When click on aby product title, show product detail with comparison table below
screen shot 2560-04-11 at 1 49 10 pm

### CODE REVIEW

**Comment by birdylockstep** [comment]
@sasitornamy Please review

File: app/views/products/show.html.erb [hunk 1]
```diff
@@ -1,28 +1,60 @@
...
+        <%= number_to_currency(@product.current_price, precision: 2) %>
+        <% if @product.original_price.present? %>
+          <span class="small text-danger"><s><%= number_to_currency(@product.original_price, precision: 2) %></s></span>
+        <% end %>
+      </h4>
+      <h5 class="my-4">Rating: <%= @product.average_rating %></h5>
+      <h5 class="my-4">
+        <span class="badge badge-success">In stock</span>
+      </h5>
+      <h5 class="my-4">from: <%= @product.retailer_name %></h5>
```
Comment (sasitornamy) [inline app/views/products/show.html.erb]: This should be extracted from manufacturer instead, right?

File: app/views/products/_list.html.erb [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
+<% @products.in_groups_of(4) do |rows| %>
+  <div class="row my-3">
+    <% rows.each do |product| %>
+      <% next unless product.present? %>
+      <div class="col-3 product" id="product-#{product.id}">
+        <div class="text-center p-2 product-box">
           <% image_path = (product.image_url) ? product.image_url : 'placeholder.png' %>
-          <%= link_to product_path(product) do %>
-            <%= image_tag(image_path, width: 100) %>
+          <%= image_tag(image_path, height: '60', class: 'mt-1') %>
```
Comment (sasitornamy) [inline app/views/products/_list.html.erb:8]: image
Comparing to tressle.com, I think the bigger image would be more beautiful ja.

File: app/views/products/index.html.erb [hunk 3]
```diff
@@ -26,12 +16,12 @@
...
-        <%= link_to products_path(sort: 'price', sort_direction: 'ASC', query: params[:query]),
-          class: "dropdown-item #{'active' if @sort == 'price' && @sort_direction == 'ASC'}" do %>
+        <%= link_to products_path(sort: 'current_price', sort_direction: 'ASC', query: params[:query]),
+          class: "dropdown-item #{'active' if @sort == 'current_price' && @sort_direction == 'ASC'}" do %>
           <%= fa_icon('sort-numeric-asc') %> Price
         <% end %>
-        <%= link_to products_path(sort: 'price', sort_direction: 'DESC', query: params[:query]),
-          class: "dropdown-item #{'active' if @sort == 'price' && @sort_direction == 'DESC'}" do %>
+        <%= link_to products_path(sort: 'current_price', sort_direction: 'DESC', query: params[:query]),
+          class: "dropdown-item #{'active' if @sort == 'current_price' && @sort_direction == 'DESC'}" do %>
```
Comment (sasitornamy) [inline app/views/products/index.html.erb]: I don't think "sort by" is working correctly, if I click on any category link, and then sort by price, I think it should sort only the products belonging to that category.
However, in the mockup design, there is no sort by feature, maybe remove it for now.

File: app/views/application/_navbar.html.erb [hunk 4]
```diff
@@ -2,7 +2,7 @@
   <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarNavDropdown" aria-controls="navbarNavDropdown" aria-expanded="false" aria-label="Toggle navigation">
     <span class="navbar-toggler-icon"></span>
   </button>
-  <h1 class="navbar-brand mb-0"> TRESSLE COMPARE </h1>
+  <a class="navbar-brand" style="font-weight: 600;" href="<%= root_path %>">TRESSLE COMPARE</a>
```
Comment (sasitornamy) [inline app/views/application/_navbar.html.erb]: actually we have a navbar.scss file for this, should we move this inline style into that file?

File: spec/features/product_page_spec.rb [hunk 5]
```diff
@@ -3,13 +3,13 @@
...
     before do
-      @product = create(:retailer_product, current_price: 1000)
+      @product = create(:product, current_price: 1000)
     end
     scenario "user can navigate to a retailer's product site" do
       visit products_path
       expect(page).to have_content 'Study Desk'
       click_on 'Study Desk'
-      expect(page).to have_link('$1,000.00', href: @product.url)
+      expect(page).to have_content @product.description
```
Comment (sasitornamy) [inline spec/features/product_page_spec.rb]: Oh..this scenario is about a user can click on a price link to navigate to external retailer product page, so in this page, should change to 'click on SHOP or BUY button' to get there.

**Review by sasitornamy — CHANGES_REQUESTED** [review]
good job so far Bird! just minor changes requested ja.

**Comment by birdylockstep** [comment]
@sasitornamy as per discussion, please review again

**Pull request merged**